// !DIAGNOSTICS: -UNUSED_PARAMETER
open class B(val prop: Int)
class A : B {
    constructor(x: Int, y: Int = x <!AMBIGUITY!>+<!> <!UNRESOLVED_REFERENCE!>prop<!> <!UNRESOLVED_REFERENCE!>+<!> <!INSTANCE_ACCESS_BEFORE_SUPER_CALL!>this<!>.<!UNRESOLVED_REFERENCE!>prop<!> <!UNRESOLVED_REFERENCE!>+<!> super.<!UNRESOLVED_REFERENCE!>prop<!>) :
        super(x <!AMBIGUITY!>+<!> <!UNRESOLVED_REFERENCE!>prop<!> <!UNRESOLVED_REFERENCE!>+<!> <!INSTANCE_ACCESS_BEFORE_SUPER_CALL!>this<!>.<!UNRESOLVED_REFERENCE!>prop<!> <!UNRESOLVED_REFERENCE!>+<!> super.<!UNRESOLVED_REFERENCE!>prop<!>)
}
